packages/clerk-js/src/ui/elements/CodeControl.tsx

```diff
@@ -274,40 +288,92 @@ export const OTPCodeControl = React.forwardRef<{ reset: any }>((_, ref) => {
     }
   };
 
+  // Handle hidden input changes (for password manager autofill)
+  const handleHiddenInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replace(/\D/g, '').slice(0, length);
+    const newValues = value.split('').concat(Array.from({ length: length - value.length }, () => ''));
+    setValues(newValues);
+
+    // Focus the appropriate visible input
+    if (value.length > 0) {
+      focusInputAt(Math.min(value.length - 1, length - 1));
+    }
+  };
+
   const centerSx = centerAlign ? { justifyContent: 'center', alignItems: 'center' } : {};
 
   return (
-    <Flex
-      isLoading={isLoading}
-      hasError={feedbackType === 'error'}
-      elementDescriptor={descriptors.otpCodeFieldInputs}
-      gap={2}
-      sx={t => ({ direction: 'ltr', padding: t.space.$1, marginLeft: `-${t.space.$1}`, ...centerSx })}
-    >
-      {values.map((value, index: number) => (
-        <SingleCharInput
-          elementDescriptor={descriptors.otpCodeFieldInput}
-          key={index}
-          value={value}
-          onClick={handleOnClick(index)}
-          onChange={handleOnChange(index)}
-          onKeyDown={handleOnKeyDown(index)}
-          onInput={handleOnInput(index)}
-          onPaste={handleOnPaste(index)}
-          id={`digit-${index}-field`}
-          ref={node => (refs.current[index] = node)}
-          autoFocus={index === 0 || undefined}
-          autoComplete='one-time-code'
-          aria-label={`${index === 0 ? 'Enter verification code. ' : ''}Digit ${index + 1}`}
-          isDisabled={isDisabled || isLoading || disabled || feedbackType === 'success'}
-          hasError={feedbackType === 'error'}
-          isSuccessfullyFilled={feedbackType === 'success'}
-          type='text'
-          inputMode='numeric'
-          name={`codeInput-${index}`}
-        />
-      ))}
-    </Flex>
+    <Box sx={{ position: 'relative' }}>
+      {/* Hidden input for password manager compatibility */}
+      <Input
+        ref={hiddenInputRef}
+        type='text'
+        autoComplete='one-time-code'
+        data-otp-hidden-input
+        inputMode='numeric'
+        pattern={`[0-9]{${length}}`}
+        minLength={length}
+        maxLength={length}
+        spellCheck={false}
+        aria-hidden='true'
+        tabIndex={-1}
+        onChange={handleHiddenInputChange}
+        onFocus={() => {
+          // When password manager focuses the hidden input, focus the first visible input
+          focusInputAt(0);
+        }}
+        sx={theme => ({
+          position: 'absolute',
+          left: '-9999px',
+          width: `calc(1px + ${passwordManagerOffset}px)`,
+          height: '1px',
+          opacity: theme.opacity.$hidden,
+          pointerEvents: 'none',
+          clipPath: `inset(0 ${passwordManagerOffset}px 0 0)`,
```

Comment:
Would the visuallyHidden util work here? or does this need to have the passwordManagerOffset even when its never visible? https://github.com/clerk/javascript/blob/main/packages/clerk-js/src/ui/styledSystem/common.ts#L193-L202

Reply:
Replaced with the util!